criu/flog.c

```diff
@@ -150,12 +153,30 @@ int flog_close(int fdout)
 	return 0;
 }
 
-int flog_encode_msg(int fdout, unsigned int nargs, unsigned int mask, const char *format, ...)
+int flog_encode_msg(int loglevel, unsigned int nargs, unsigned int mask, const char *format, ...)
```

Comment:
Looking at this function, I think it makes sense to integrate common functionality with `vprint_on_level()` into some new function. At this moment with binary log you won't know when the logging is set up by criu and text logging uses an early log for such case which here is omitted and stdout/stderr may get some binary message out.
Also, it may be worth to still have LOG_TIMESTAMP functionality.

Reply:
Having one common function would also take the burden of defining all pr_*() macro twice for flog_encode() and for print_on_level()

Reply:
As per my understanding, you are asking me to eliminate the need for print_on_level() and transfer all log calls to flog_encode() for all the architectures. The issue arises with the centos architecture, it doesn’t recognize _Generic and throws error upon encountering it. This is the main reason for having 2 definitions of pr_*() marcos.
If I try to have an intermediate function to segregate the request for binary logging compatible architectures and incompatible architectures, the issue is sending the arguments from one function to another. The flog needs all the arguments in its original form and I have not been able to find a way to do it from one function to another.
Also, I had faced issues due to the early logging but hadn’t realized then that the messages in the terminal were being printed because of this. I will try to introduce an early logging facility, similar to the normal printer, for the binary printer too.
I also plan to integrate LOG_TIMESTAMP and PID and make the difference between the normal printer and the binary printer, after decoding, minimal. 
Please let me know if I misunderstood and you had meant something different.

Reply:
Hmm, I'll try to help you with this when you'll have the next version of PR.
At this moment, instead of redefining all pr_*() functions, could you try to rather redefine `print_on_level()` as a macro under `ifdef CONFIG_HAS_FLOG`?